x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PushDownAndCombineLimits.java

```diff
@@ -43,7 +43,7 @@ public LogicalPlan rule(Limit limit, LogicalOptimizerContext ctx) {
                 || unary instanceof Project
                 || unary instanceof RegexExtract
                 || unary instanceof Enrich
-                || unary instanceof Completion) {
+                || unary instanceof InferencePlan<?>) {
```

Comment:
looks like we are also making sure RERANK is part of `PushDownAndCombineLimits` which is fine - but maybe we should call it out in the PR description